Handle MongoDB BsonDiscriminator (#7073)
quarkusio/quarkus: docs/src/main/asciidoc/mongodb.adoc

}
----

== The POJO Codec

> Quarkus initializes the link:http://mongodb.github.io/mongo-java-driver/3.12/bson/pojos[MongoDB POJO Codec], this codec is capable of automatically working with POJOs so in many cases,

Reviewer: I don't think this sentence provides any useful information. I propose we drop it and add the link to the next sentence

Author: I think we need to tell the user that the POJO codec is initialized by Quarkus in some way, and in the previous section we talk about creating custom codec so I wanted to make it clear that this is not mandatory as in most cases the pojo codec will works OOTB.

Reviewer: Then perhaps the 2 sentences can be combined like so:

```
The POJO Codec provides a set of annotations that enable the customization of the way a POJO is mapped to a MongoDB collection and This codec is initialized automatically by Quarkus
```

you will be able to avoid the creation of a specific codec.

> The POJO Codec provides a rich set of features to customize the way your POJO is mapped to a collection using a set of annotations.

Reviewer: ```suggestion
The POJO Codec provides a set of annotations that enable the customization of the way a POJO is mapped to a MongoDB collection.
```

> One of these annotations is the `@BsonDiscriminator` annotation that allows to store in a single collection multiple Java types by adding

Reviewer: ```suggestion
One of these annotations is the `@BsonDiscriminator` annotation that allows to storage multiple Java types in a single MongoDB collection by adding
```

a discriminator field inside the document. It can be useful when working with abstract types or interfaces.

> Quarkus will automatically register all the classes annotated with `@BsonDiscriminator` inside the POJO codec, so you don't have to do it by yourself.

Reviewer: ```suggestion
Quarkus will automatically register all the classes annotated with `@BsonDiscriminator` with the POJO codec.
```
